onset_maproom/calc.py

```diff
@@ -53,6 +332,228 @@ def water_balance(
+    in which case evapotranspiration is scaled by Kc
+    If not, Kc is 1
+    If crop dependent, crops need be planted at a planting date
+    If planting date (p_d) is not given,
+    it is estimated by planting_date function (dependent on sm
+    and on sm_threshold that is set by default at 20 mm)
+    Finally, an additional stress scale factor, Ks,
+    can be triggered by definining rho
+    then Ks is estimated by single_stress_coeff
+    Then the outputs of the soil plant water balance are:
+    soil moisture,
+    runoff,
+    effective precipitation,
+    evapotranspiration (input) broadcasted against peffective,
+    planting date (if it exists),
+    reduced (or actual) evapotranspiration,
+    drain
+    """
+    # Start water balance ds with runoff
+    if runoff is None:
+        runoff = xr.zeros_like(daily_rain).rename("runoff")
+        runoff.attrs = dict(description="Runoff", units="mm")
+    water_balance = xr.Dataset().merge(runoff)
+    # Compute Effective Precipitation
+    peffective = (daily_rain - water_balance.runoff).rename("peffective")
+    peffective.attrs = dict(description="Effective Precipitation", units="mm")
+    water_balance = water_balance.merge(peffective)
+    # Additional vars and coords
+    et = (et * xr.ones_like(water_balance.peffective)).rename("et")
+    water_balance = water_balance.merge(et)
+    if kc_params is not None:
+        for adim in kc_params.dims:
+            for thedims in water_balance.dims:
+                if adim != thedims and adim != "kc_periods":
+                    water_balance[adim] = kc_params[adim]
+    if p_d is not None:
+        for adim in p_d.dims:
+            for thedims in water_balance.dims:
+                if adim != thedims and adim != time_coord:
+                    water_balance[adim] = p_d[adim]
+    # Get time_coord info
+    time_coord_size = water_balance.peffective[time_coord].size
+    # Intializing sm, et_crop and et_crop_red
+    # Creating variables and broadcasting everybody
+    soil_moisture = xr.full_like(
+        water_balance.peffective, np.nan
+    ).rename("soil_moisture")
+    soil_moisture.attrs = dict(description="Soil Moisture", units="mm")
+    water_balance = water_balance.merge(soil_moisture)
+    et_crop = xr.full_like(
+        water_balance.peffective, np.nan
+    ).rename("et_crop")
+    et_crop.attrs = dict(description="Crop Evapotranspiration", units="mm")
+    water_balance = water_balance.merge(et_crop)
+    (water_balance,) = xr.broadcast(water_balance)
+    water_balance["soil_moisture"] = water_balance.soil_moisture.copy()
+    water_balance["et_crop"] = water_balance.et_crop.copy()
+    # Give time dimension to sminit
+    sminit0 = xr.full_like(
+        water_balance.soil_moisture
+        .isel({time_coord: 0}).expand_dims(dim=time_coord),
+        sminit,
+    )
+    sminit0 = sminit0.assign_coords(
+        {time_coord: sminit0[time_coord] - np.timedelta64(1, "D")}
+    )
```

Comment:
I'm rereading this comment that I made last week (but didn't submit until yesterday), and it doesn't make sense to me. I suspect `T=10` was a typo, but I'm not sure what I meant to type. I still think the `isel` is a bit confusing, since you're not actually selecting values from `soil_moisture`, only from its time dimension. Maybe something like
```
t0 = water_balance['soil_moisture']['T'][0] - np.timedelta64(1, 'D')
sminit0 = xr.DataArray(coords=dict(T=[t0]), data=[sminit]))
```